src/mkappimage/mkappimage.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"github.com/probonopd/go-appimage/internal/helpers"
+	"github.com/urfave/cli/v2"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+
+// listFilesInAppImage lists the files in the AppImage, similar to
+// the ls command in UNIX systems
+func listFilesInAppImage(path string) {
+	appimage := NewAppImage(path)
+	err := appimage.ShowContents(false)
+	if err != nil {
+		log.Fatal("Failed to list files: error code:", err)
+	}
+
+}
+
+// listLongFilesInAppImage lists all the files in the appimage
+// similar to `ls -al`. It shows the file permissions also
+func listLongFilesInAppImage(path string) {
+	appimage := NewAppImage(path)
+	err := appimage.ShowContents(true)
+	if err != nil {
+		log.Fatal("Failed to list files: error code:", err)
+	}
+
+}
+
+
+// bootstrapMkAppImage is a function which converts cli.Context to
+// string based arguments, checks if all the files
+// provided as arguments exists. If yes add the current path to PATH,
+// check if all the necessary dependencies exist,
+// finally check if the provided argument, AppDir is a directly.
+// Call GenerateAppImage with the converted arguments
+// 		Args: c: cli.Context
+func bootstrapMkAppImage(c *cli.Context) error {
+
+	// check if the number of arguments are stictly 1, if not
+	// return
+	if c.NArg() < 1 {
```

Comment:
```suggestion
	if c.NArg() != 1 {
```

If only the first positional argument is ever used, please fail if the user provides more than one accidentally.

Reply:
Improved it.